### PR TITLE
Fix pod status logic to account for init containers

### PR DESCRIPTION
**Pull request by TheRealJon**
Fixes https://github.com/openshift/console/issues/107

Add logic to `frontend/public/module/k8s/pods.ts` to account for init containers. Pod list and detail pages will now show init container failure/waiting states in pod status field.

### CODE REVIEW

File: frontend/public/module/k8s/pods.ts [hunk 1]
```diff
@@ -177,26 +176,41 @@ export const podPhase = (pod): PodPhase => {
...
+      if (terminated && terminated.reason) {
+        phase = `Init${terminated.reason}`;
+      } else if (waiting && waiting.reason && waiting.reason !== 'PodInitializing') {
+        phase = `Init${waiting.reason}`;
       }
+      initializing=true;
+    });
+  }
+
+  if (!initializing) {
```
Comment (spadgett) [inline frontend/public/module/k8s/pods.ts]: You can combine these ifs

```js
if (!initializing && !_.isEmpty(pod.status.containerStatuses)) {
```

(or switch to `_.each`)

File: frontend/public/module/k8s/pods.ts [hunk 2]
```diff
@@ -177,26 +176,41 @@ export const podPhase = (pod): PodPhase => {
...
+  let phase = pod.status.phase || pod.status.reason;
 
-  if (pod.status.containerStatuses) {
-    pod.status.containerStatuses.forEach(function(container) {
-      if (container.state.waiting && container.state.waiting.reason) {
-        ret = container.state.waiting.reason;
-      } else if (container.state.terminated && container.state.terminated.reason) {
-        ret = container.state.terminated.reason;
+  if (pod.status.initContainerStatuses) {
+    pod.status.initContainerStatuses.forEach(function(container) {
```
Comment (spadgett) [inline frontend/public/module/k8s/pods.ts]: If you use `_.each`, you can skip the check for existence.

```js
_.each(pod.status.initContainerStatuses, container => {
```

Comment (TheRealJon) [reply frontend/public/module/k8s/pods.ts]: Yeah, I like that better

File: frontend/public/module/k8s/pods.ts [hunk 3]
```diff
@@ -177,26 +176,37 @@ export const podPhase = (pod): PodPhase => {
...
       }
-      // kubectl has code here that populates the field if
-      // terminated && !reason, but at this writing there appears to
-      // be no codepath that will produce that sort of output inside
-      // of the kubelet.
     });
+    // kubectl has code here that populates the field if
+    // terminated && !reason, but at this writing there appears to
+    // be no codepath that will produce that sort of output inside
+    // of the kubelet.
```
Comment (spadgett) [inline frontend/public/module/k8s/pods.ts]: nit: comment probably shouldn't move :)

**Comment by spadgett** [comment]
Thanks @TheRealJon 👍 

/lgtm

**Pull request merged**